Cut the first few paragraphs down, by quite a lot (#52)
pavpanchekha/emberfox: book/browsersandweb.md

prev: preliminaries
...

The web browser, and the web[^theweb] more broadly, is a marvel. Every
year it expands its reach to more and more of what we do with
computers. It now goes far beyond its original use for document-based
information sharing; many people now spend their entire day in a
browser, not using a single other application.

[^theweb]: Broadly defined, the web is the interlinked network (“web”) of
    [web pages](https://en.wikipedia.org/wiki/Web_page) on the Internet.

Nowadays, desktop applications are often built and delivered as _web
apps_: websites[^website] used in similar ways to installed
applications. And on mobile devices even native apps often use _web

> views_ that embed a browser render parts of the application UI.

Reviewer: ```suggestion
views_ that embed a browser to render parts of the application UI.
```

Perhaps in the future mobile devices will, like desktop computers,

> mostly a container for web apps.

Reviewer: ```suggestion
mostly be a container for web apps.
```

[^website]: You probably already know what a website is. This is one.
    If not, [see here](https://en.wikipedia.org/wiki/Website).

[^hybrid]: The fraction of these _hybrid apps_ that are web content is
    also likely increasing over time.

Yet this critical piece of software is built on a few simple, yet
revolutionary, concepts that present a vision of the future of
software and information: open, decentralized and safe computing;
declarative UIs; hyperlinks; and user agents[^useragent]. This vision

> is neither simple nor obvious, even if it seems that way in hindsight:

Reviewer: Here you changed the argument to emphasize the other technologies that made the web possible, but skipped the part about prior research into information systems. Was that on purpose?

Author: Yeah, I think I did not do this well. The real thought here was to keep the focus on browsers, whereas this paragraph changes focus to the web and concepts that aren't directly related to it. I tried swapping parts between this and the next paragraph, and that didn't really work, so I kind of ended up with this.

The idea of focusing on browsers still seems good but I think you can do a better job than I did.

Reviewer: Ok. Revert this part or I'll try to patch in parts of your PR?

Reviewer: You make a good point about the change of focus. Here is another take on resolving this issue. Better?

https://github.com/pavpanchekha/emberfox/pull/53

prev: preliminaries
...

The web browser, and the web[^theweb] more broadly, is a marvel. Every
year it expands its reach to more and more of what we do with
computers. It now goes far beyond its original use for document-based
information sharing; many people now spend their entire day in a

> browser, not using a single other application.

Reviewer: ```suggestion
browser, not using a single other application!
```

[^theweb]: Broadly defined, the web is the interlinked network (“web”) of
    [web pages](https://en.wikipedia.org/wiki/Web_page) on the Internet.

Nowadays, desktop applications are often built and delivered as _web
apps_: websites[^website] used in similar ways to installed
applications. And on mobile devices even native apps often use _web
views_ that embed a browser to render parts of the application UI.
Perhaps in the future mobile devices will, like desktop computers,
mostly be a container for web apps.

> [^website]: You probably already know what a website is. This is one.

Reviewer: ```suggestion
[^website]: You probably already know what a website is. You're  probably reading this book on one right now!
```

    If not, [see here](https://en.wikipedia.org/wiki/Website).

[^hybrid]: The fraction of these _hybrid apps_ that are web content is
    also likely increasing over time.

The basis of this critical piece of software is the web. And the web
itself is built on a few simple, yet revolutionary, concepts; concepts
that that together present a vision of the future of software and
information. Among them are open, decentralized and safe computing; a
declarative document model for describing UIs; hyperlinks; and the

> User Agent concept[^useragent]. Since the browser makes the web real,

Reviewer: ```suggestion
User Agent[^useragent]. Since the browser makes the web real,
```

all these concepts form the core structure of the browser code itself.

[^useragent]: The User Agent is a way to view the computer, or
    software within the computer, as a trusted assistant and advocate.

This vision of the web is neither simple nor obvious; it is the result

> ofexperimentation and research reaching back to nearly the beginning

Reviewer: ```suggestion
of experimentation and research reaching back to nearly the beginning
```

> of computing. And the web _also_ needed rich computer displays,

Reviewer: ```suggestion
of computing. And of course the web _also_ needs rich computer displays,
```

powerful UI-building libraries, fast consumer networks, and sufficient
CPU power and information storage capacity. As so often happens, the
web has many predecessors but only took its modern form in the late

> 1990s.

Reviewer: ```suggestion
1980s, once all those technologies were available.
```
